### PR TITLE
no this

### PR DESCRIPTION
**Pull request by sergey-shandar**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -34,44 +34,41 @@ export interface IterableEx<T> extends Iterable<T> {
...
 
-// tslint:disable-next-line:no-class
-class IterableImpl<T> implements IterableEx<T> {
-    public readonly [Symbol.iterator]: () => Iterator<T>
-    constructor(createIterator: () => Iterator<T>) {
-        // tslint:disable-next-line:no-expression-statement
-        this[Symbol.iterator] = createIterator
+export const iterable = <T>(createIterator: () => Iterator<T>): IterableEx<T> => {
+    const property = <P extends readonly unknown[], R>(f: (self: Iterable<T>, ...p: P) => R) =>
+        (...p: P) => f(result, ...p)
```
Comment (timotheeguerin) [inline src/index.ts]: can you rename `result` to something slightly more descriptive? also does the `extends unkown[]` really bring anything here?

Comment (sergey-shandar) [reply src/index.ts]: yes, TypeScript requires that `P` is an array

Comment (timotheeguerin) [reply src/index.ts]: oh I see

Comment (sergey-shandar) [reply src/index.ts]: The name `result` is a convention, similar to `this`. The `result` const is a result of the function.

**Pull request merged**